pkg/generate/code/compare.go

```diff
@@ -564,7 +564,8 @@ func CompareStruct(
 		var compareConfig *ackgenconfig.CompareFieldConfig
 		// memberFieldPath contains the field path along with the prefix cfg.PrefixConfig.SpecField + "." hence we
 		// would need to substring to exclude cfg.PrefixConfig.SpecField + "." to get correct field config.
-		fieldConfig := fieldConfigs[memberFieldPath[len(cfg.PrefixConfig.SpecField) + 1 :len(memberFieldPath)]]
+		specFieldLen := len(strings.TrimPrefix(cfg.PrefixConfig.SpecField, "."))
```

Comment:
This fix makes sense, i still wonder what part of elasticache/memorydb delta generated code was broken? can we add some unit tests? :)

Reply:
I added a new unit test :)